app/src/org/commcare/dalvik/activities/TemplatePrinterActivity.java

```diff
@@ -0,0 +1,283 @@
+                                finish();
+                            }
+                        }
+                );
+        dialogBuilder.show();
+    }
+
+    /**
+     * Shows an alert dialog about the status of the print job.
+     * Activity will quit upon exiting the dialog.
+     *
+     * @param msg String message that should be shown on the alert
+     */
+    public void showAlertDialog(String msg) {
+        AlertDialog.Builder alert = new AlertDialog.Builder(this);
+        alert.setTitle(msg);
+        alert.setCancelable(false);
+        alert.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.dismiss();
+                finish();
+            }
+        });
+
+        alert.show();
+    }
+
+    /**
+     * Prepares a WebView of the html document generated by this activity, which can then be
+     * printed by the Android print framework
+     *
+     * Source: https://developer.android.com/training/printing/html-docs.html
+     */
+    private void doHtmlPrint() {
+        // Create a WebView object specifically for printing
+        WebView webView = new WebView(this);
+        webView.setWebViewClient(new WebViewClient() {
+
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                return false;
+            }
+
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                createWebPrintJob(view);
+                mWebView = null;
+            }
+        });
+        try {
+            String htmlDocString = TemplatePrinterIOUtil.readStringFromFile(outputPath);
+            webView.loadDataWithBaseURL(null, htmlDocString, "text/HTML", "UTF-8", null);
+
+            // Keep reference to WebView object until PrintDocumentAdapter is passed to PrintManager
+            mWebView = webView;
+        } catch (IOException e) {
+            showErrorDialog("Could not read from generated HTML doc in order to print");
+        }
+
+    }
+
+    /**
+     * Starts a print job for the given WebView
+     *
+     * Source: https://developer.android.com/training/printing/html-docs.html
+     *
+     * @param v the WebView to be printed
+     */
+    @TargetApi(Build.VERSION_CODES.KITKAT)
+    private void createWebPrintJob(WebView v) {
+
+        // Get a PrintManager instance
+        PrintManager printManager = (PrintManager) getSystemService(Context.PRINT_SERVICE);
+
+        // Get a print adapter instance
+        PrintDocumentAdapter printAdapter = v.createPrintDocumentAdapter();
+
+        // Create a print job with name and adapter instance
+        printJob = printManager.print(jobName, printAdapter,
+                new PrintAttributes.Builder().build());
+        jobStarted = true;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (jobStarted) {
```

Comment:
check here for the pattern you're looking for:

http://stackoverflow.com/questions/30742051/android-printmanager-get-callback